router/core/graphql_prehandler.go

```diff
@@ -315,11 +317,29 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 			body:             body,
 		})
 		if err != nil {
-			requestContext.error = err
+			outputErr := err
+			var httpErr ApolloCompatibleHttpError
+			var reportErr ReportError
+			if h.executor.Resolver.Options().ApolloCompatibilityReplaceInvalidVariableError && errors.As(err, &httpErr) {
```

Comment:
`httpErr` is returned for many reasons not just for invalid variables.

Reply:
This an apolloCompatibleHttpError. It has a different form to a typical http error. I think the variable name is confusing.

Reply:
Oh yeah, we use the same naming for general http errors.